crates/runtime/src/instance/allocator/pooling/index_allocator.rs

```diff
@@ -199,29 +121,15 @@ enum AllocMode {
 
 impl IndexAllocator {
     /// Create the default state for this strategy.
-    pub fn new(strategy: PoolingAllocationStrategy, max_instances: usize) -> Self {
-        let ids = (0..max_instances).map(|i| SlotId(i)).collect::<Vec<_>>();
-        // Use a deterministic seed during fuzzing to improve reproducibility of
-        // test cases, but otherwise outside of fuzzing use a random seed to
-        // shake things up.
-        let seed = if cfg!(fuzzing) {
-            [0; 32]
-        } else {
-            rand::thread_rng().gen()
-        };
-        let rng = SmallRng::from_seed(seed);
+    pub fn new(max_instances: u32, max_unused_warm_slots: u32) -> Self {
```

Comment:
Adding this assertion would require adding validation to the `PoolingAllocatorConfig` in `wasmtime` as well to provide a better error message than tripping the assertion. Thinking about that though I think it may not be worth it since it's not really a problem if `max_unused_warm_slots` is bigger than the number of slots. It's a bit silly but it can also perhaps be helpful to always pass a large value here to say "always keep everything warm"